### PR TITLE
Provide options for legacy add-ons

### PR DESCRIPTION
**Pull request by cschroeter**
Solves https://github.com/home-assistant/hassio/issues/805

### CODE REVIEW

File: hassio/docker/addon.py [hunk 1]
```diff
@@ -88,6 +88,12 @@ def environment(self):
                 'ALSA_INPUT': self.addon.audio_input,
             })
 
+        # Provide options for legacy add-ons
+        if self.addon.legacy:
+            addon_env.update({
+                **self.addon.options
+            })
+
```
Comment (pvizeli) [inline hassio/docker/addon.py]: ENV support only one depth. We should make sure, that there is no nested struct inside. `**{value: key in value, key for options.items() if isinstance(key, str, int)}`

**Comment by cschroeter** [comment]
Mh if you flag your add-on with legacy (non-hassio) the options you pass may not be nested. I would rather check this in the validation part of the add-on than in the Docker part. 
What you say?

**Comment by pvizeli** [comment]
Sure, that was my second opinion.

**Comment by pvizeli** [comment]
https://github.com/home-assistant/hassio/blob/dev/hassio/addons/validate.py#L111

You can create it like `vol.Schema(vol.All({...}, _validate_legacy, ...)` and the function `validate_legacy` trow a `vol.Invalid` if legacy is set but the options is not valid.

**Comment by cschroeter** [comment]
Jus because I set the flag _legacy_ my add-ons with nesteds options won't start?  Doesn't legacy add-ons also recieve the volumen with those options.json? It feels like a misuse of this flag.

I also don't feel comfortable passing those options as Docker env with out further notice. Maybe we could introduce something like a new flag - "pass options as docker env". What you say? If you don't agree I would vote to just pass the first level of options instead of deny a user that flags his add-on _legacy_  to use nested options

File: hassio/docker/addon.py [hunk 2]
```diff
@@ -88,6 +88,15 @@ def environment(self):
                 'ALSA_INPUT': self.addon.audio_input,
             })
 
+        # Provide options for legacy add-ons
+        if self.addon.legacy:
+            for key, value in self.addon.options.items():
+                if isinstance(value, (int, str)):
+                    addon_env.update({key: value})
```
Comment (pvizeli) [inline hassio/docker/addon.py]: `addon_env[key] = value`

**Comment by pvizeli** [comment]
Thanks

**Pull request merged**